src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseHeaderMap.cs

```diff
@@ -0,0 +1,256 @@
+                        reader?.SkipValue();
+                        break;
+                    case KnownHeaders.Crit:
+                        throw new NotSupportedException(); // TODO
+                        //if (reader == null)
+                        //{
+                        //    throw new InvalidOperationException(SR.Format(SR.CoseHeaderMapHeaderDoesNotAcceptSpecifiedValue, label));
+                        //}
+
+                        //int? length = reader.ReadStartArray();
+                        //if (length.GetValueOrDefault() < 1)
+                        //{
+                        //    throw new InvalidOperationException(SR.Format(SR.CoseHeaderMapHeaderDoesNotAcceptSpecifiedValue, label));
+                        //}
+
+                        //for (int i = 0; i < length; i++)
+                        //{
+                        //    CborReaderState state = reader.PeekState();
+                        //    if (state != CborReaderState.NegativeInteger &&
+                        //        state != CborReaderState.UnsignedInteger &&
+                        //        state != CborReaderState.TextString)
+                        //    {
+                        //        throw new InvalidOperationException(SR.Format(SR.CoseHeaderMapHeaderDoesNotAcceptSpecifiedValue, label));
+                        //    }
+                        //    reader.SkipValue();
+                        //}
+                        //reader.ReadEndArray();
+                        //break;
+                    case KnownHeaders.CounterSignature:
+                        throw new NotSupportedException(); // TODO
+                    case KnownHeaders.ContentType:
+                        if (initialState != CborReaderState.TextString &&
+                            initialState != CborReaderState.UnsignedInteger)
+                        {
+                            throw new InvalidOperationException(SR.Format(SR.CoseHeaderMapHeaderDoesNotAcceptSpecifiedValue, label));
+                        }
+                        reader?.SkipValue();
+                        break;
+                    case KnownHeaders.Kid:
+                    case KnownHeaders.IV:
+                    case KnownHeaders.PartialIV:
+                        if (initialState != CborReaderState.ByteString)
+                        {
+                            throw new InvalidOperationException(SR.Format(SR.CoseHeaderMapHeaderDoesNotAcceptSpecifiedValue, label));
+                        }
+                        reader?.SkipValue();
+                        break;
+                    default:
+                        reader?.SkipValue();
+                        break;
+                }
+            }
+        }
+
+        internal byte[] Encode(bool mustReturnEmptyBstrIfEmpty = false)
+        {
+            if (_headerParameters.Count == 0 && mustReturnEmptyBstrIfEmpty)
+            {
+                return s_emptyBstrEncoded;
```

Comment:
I can't easily tell from the PR... does this instance ever propagate out of the public surface area?  If yes, we should be thoughtful about it being a singleton that a consumer could modify.

Reply:
It won't leak as it will be re-encoded by a `CborWriter` before returning.